src/GafferScene/IECoreGLPreview/Renderer.cpp

```diff
@@ -213,28 +217,52 @@ class OpenGLAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 			return m_state.get();
 		}
 
-		const IECoreGL::Renderable *visualisation() const
+		const IECoreGL::Renderable *visualisation( VisualisationType type ) const
 		{
-			return m_visualisation.get();
+			const auto &it = m_visualisations.find( type );
```

Comment:
And I guess this is where `array` would have an advantage over `flat_map` in terms of syntax.